test/games/strategy/triplea/ui/TestRoute.java

```diff
@@ -0,0 +1,64 @@
+package games.strategy.triplea.ui;
```

Comment:
Good start on test cases.

Looking at these, the coverage does not appear very deep. Second thing I notice is that a lot of helper / protected methods are being tested. That is one way to test things, but increasing scope of a helper method so you can test it is a warning sign. In specific it then does not help you when you need to test the clients of those helper methods, you'll wind up with redundant testing.

I'd like to see us avoid the test of the helper methods.

One way to do this:
- create a package that will contain all the drawing logic / classes
- for each protected helper method, make a class out of it.
- Then test each class individually, and use mocks to stub out the dependencies. Then for example, you could mock out the 'getIndex()' functionality, and then your test could control the values coming from that method. Basically anywhere you have a protected method, you probably should have a mock.
- Then once you have things decomposed, your top level algorithm will actually be high level, and it'll be resistant to change since the details will be hidden from it, so it would only be a helper class that you would actually have to change if you update the drawing algorithm.